AstronomyTests/MarsRoverClientTests.swift

```diff
@@ -0,0 +1,215 @@
+//
+//  MarsRoverClientTests.swift
+//  MarsRoverClientTests
+//
+//  Created by Hunter Oppel on 5/18/20.
+//  Copyright © 2020 Lambda School. All rights reserved.
+//
+
+import XCTest
+@testable import Astronomy
+
+/*
+ Does decoding work?
+ Does decoding fail when given bad data?
+ Does it build the correct URL?
+ Does it build the correct URLRequest?
+ Are the search results saved properly?
+ Is the completion handler called when data is good?
+ Is the completion handler called when data is bad?
+ Is the completion handler called when networking fails?
+ */
+
+class MarsRoverClientTests: XCTestCase {
+    
+    private let fakeRover = MarsRover(name: "Friend",
+                                      launchDate: Date(),
+                                      landingDate: Date() + 2,
+                                      status: .active,
+                                      maxSol: 1,
+                                      maxDate: Date(),
+                                      numberOfPhotos: 16,
+                                      solDescriptions: [])
+    
+    func testForPhotosWithServer() {
+        let roverExpectation = expectation(description: "Wait for rover info")
+        
+        let controller = MarsRoverClient(networkLoader: URLSession(configuration: .ephemeral))
+        var roverInfo: MarsRover?
+        
+        controller.fetchMarsRover(named: "curiosity") { rover, error in
+            XCTAssertNil(error)
+            
+            XCTAssertNotNil(rover)
+            NSLog("✅ We got a rover!")
+            roverInfo = rover
+            roverExpectation.fulfill()
+        }
+        
+        wait(for: [roverExpectation], timeout: 5)
+        
+        let photosExpectation = expectation(description: "Wait for photos")
+        
+        controller.fetchPhotos(from: roverInfo!, onSol: 1) { photos, error in
+            XCTAssertNil(error)
+            
+            XCTAssertNotNil(photos)
+            NSLog("✅ We got photos!")
+            XCTAssertGreaterThan(photos!.count, 0)
+            photosExpectation.fulfill()
+        }
+        
+        wait(for: [photosExpectation], timeout: 5)
+    }
+    
+    func testSpeedOfTypicalRequest() {
+        measure {
+            let roverExpectation = expectation(description: "Wait for rover info")
+            
+            let controller = MarsRoverClient(networkLoader: URLSession(configuration: .ephemeral))
+            var roverInfo: MarsRover?
+            
+            controller.fetchMarsRover(named: "curiosity") { rover, error in
+                NSLog("✅ We got a rover!")
+                roverInfo = rover
+                roverExpectation.fulfill()
+            }
+            
+            wait(for: [roverExpectation], timeout: 10)
```

Comment:
Perhaps move this block of code out of the measure block, so tyou are only measuring the photos request